src/nvim/undo.c

```diff
@@ -1862,6 +1862,7 @@ static void u_doit(int startcount, bool quiet, bool do_buf_event)
     u_oldcount = -1;
   }
 
+  msg_ext_set_kind("undo");
```

Comment:
Please verify if there is no better place to this call.

Reply:
I think so yes, which makes the one you added in `u_undo_end()` redundant.
I pushed a change in which I removed it, moved the test to the umbrella kind test, and exchanged the messages for ones without the time of the change, which tends to result in flakiness (1 vs 0 seconds).

Reply:
Indeed, I see that after removal the test case passes too. Thanks!